clean/platforms/nextrequest.py

```diff
@@ -0,0 +1,363 @@
+            "folder_id": parse_qs(parsed_url.query)["folder_filter"][0],
+            "page_size": 50,
+            "doc_limit": 9950,  # Max number of accessible docs in a folder
+            "tally_field": "total_count",
+            # "document_path": "document_path",
+            "bln_page_url": "bln_page_url",
+            "bln_total_documents": "bln_total_documents",
+        }
+        line["json_url"] = (
+            f"{line['base_url']}/client/documents?sort_field=count&sort_order=desc&page_size=50&folder_filter={line['folder_id']}&page_number="
+        )
+        line["details"] = {
+            "document_path": "document_path",
+            "description": "description",
+            "count": "count",
+            "state": "state",
+            "demo": "demo",
+            "created_at": "created_at",
+            "folder_name": "folder_name",
+            "redacted_at": "redacted_at",
+            "file_extension": "file_extension",
+            "doc_date": "doc_date",
+            "id": "id",
+            "highlights": "highlights",
+            "bln_page_url": "bln_page_url",
+            "bln_total_documents": "bln_total_documents",
+        }
+
+    elif (  # BARTish type
+        len(parsed_url.path.split("/")) == 3
+        and parsed_url.path.split("/")[1] == "requests"
+    ):
+        line = {
+            "site_type": "bartish",  # Bartish type
+            "base_url": f"{parsed_url.scheme}://{parsed_url.netloc}",
+            "folder_id": urlparse(start_url).path.split("/")[2],
+            "doc_limit": 9950,  # Max number of accessible docs in a folder
+            "page_size": 25,
+            "tally_field": "total_documents_count",
+            #            "document_path": "document_scan['document_path']",
+        }
+        line["json_url"] = (
+            f"{line['base_url']}/client/request_documents?request_id={line['folder_id']}&page_number="
+        )
+        line["details"] = {
+            "document_path": "ds!document_path",
+            "bogus_asset_url": "asset_url",
+            "review_state": "review_state",
+            "review_status": "ds!review_status",
+            "severity": "ds!severity",
+            "findings": "ds!findings",
+            "file_extension": "file_extension",
+            "file_size": "ds!file_size",
+            "file_type": "ds!file_type",
+            "visibility1": "visibility",
+            "visibility2": "ds!visibility",
+            "upload_date1": "upload_date",
+            "upload_date2": "ds!upload_date",
+            "pretty_id": "ds!pretty_id",
+            "id1": "id",
+            "id2": "ds!id",
+            "document_id": "ds!document_id",
+            "request_id": "request_id",
+            "folder_name": "folder_name",
+            "subfolder_name": "subfolder_name",
+            "exempt_from_retention": "exempt_from_retention",
+            "bln_page_url": "bln_page_url",
+            "bln_total_documents": "bln_total_documents",
+        }
```

Comment:
for readability, don't break up the `line` dictionaries

```suggestion
        return {
            "site_type": "bartish",
            "base_url": f"{parsed_url.scheme}://{parsed_url.netloc}",
            "folder_id": parsed_url.path.split("/")[2],
            "doc_limit": BARTISH_DOC_LIMIT,
            "page_size": BARTISH_PAGE_SIZE,
            "tally_field": "total_documents_count",
            "json_url": f"{parsed_url.scheme}://{parsed_url.netloc}/client/request_documents?request_id={parsed_url.path.split('/')[2]}&page_number=",
            "details": {
                "document_path": "ds!document_path",
                "bogus_asset_url": "asset_url",
                "review_state": "review_state",
                "review_status": "ds!review_status",
                "severity": "ds!severity",
                "findings": "ds!findings",
                "file_extension": "file_extension",
                "file_size": "ds!file_size",
                "file_type": "ds!file_type",
                "visibility1": "visibility",
                "visibility2": "ds!visibility",
                "upload_date1": "upload_date",
                "upload_date2": "ds!upload_date",
                "pretty_id": "ds!pretty_id",
                "id1": "id",
                "id2": "ds!id",
                "document_id": "ds!document_id",
                "request_id": "request_id",
                "folder_name": "folder_name",
                "subfolder_name": "subfolder_name",
                "exempt_from_retention": "exempt_from_retention",
                "bln_page_url": "bln_page_url",
                "bln_total_documents": "bln_total_documents",
            },
         }
```

Reply:
I've got those dictionaries back together by shifting one bit of code out. Which somehow also required more mypy ignores.

Can you take a look and see if this works?

I think if we break this down further we're just making it harder to find code. Here the fingerprint type you want is identified by indents, and you can just scroll down. Otherwise we're basically repeating the structure with more if statements in a profile generator function, or we're creating multiple profile functions that do nothing except scatter responsibility further.